lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -3437,15 +3437,32 @@ LogicalResult ConvertAtenOp<AtenBroadcastToOp>::matchAndRewrite(
   // Get the result type
   auto resultType = getTypeConverter()->convertType(op.getType());
 
+  int64_t numBroadcastedDims = resultShape.size() - selfType.getRank();
```

Comment:
Can this be negative? I'd prefer to check and return here if it is,
but it would also be interesting whether torch.broadcast would fail.

Reply:
If you are sure that this can never happen (which sounds like it), then I would add this knowledge to the code in form of an `assert`.